app/javascript/components/cloud-provider/cloud-ec2-provider.schema.js

```diff
@@ -0,0 +1,83 @@
+import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
+import get from 'lodash/get';
+
+const prefix = 'ManageIQ::Providers::Amazon::CloudManager';
+
+const asyncValidate = formValues => new Promise((resolve, reject) => {
+  const azureValues = formValues[prefix] || {};
+
+  const values = {
+    button: 'validate',
+    name: get(formValues, 'name'),
+    emstype: 'ec2', // get(formValues, 'type'),
+    provider_region: get(azureValues, 'region'),
+    azure_tenant_id: get(azureValues, 'uid_ems'),
+    subscription: get(azureValues, 'subscription'),
+    zone: 'default',
+    default_url: get(azureValues, 'endpoints[0].url'),
```

Comment:
why not just `azureValues.endpoints[0].url`?

All those `get` calls are suprising, given the right side is a constant.

Reply:
They might not be inside the form state. Not all of the nested fields are required and if you don;t change it it will be `undefined`. I agree there is a lot of gets but that whole validation call will change and will be via API. Its just a placeholder for now.